### DISCUSSION TITLE
How to control frontend creation of users with role-based access control?

### DISCUSSION THREAD

**Discussion by jopfre** [supabase/supabase#8083] [Questions] [answered]
I have used the NextJs quickstart guide to get supabase and the frontend set up and everything is working well.

I have then used code from the next-js-slack-clone/full-schema.sql to add roles-based access control to the database and this also looks good.

My goal is to have a front end user management page where only users with the role of 'admin' can create, update and delete other users.

However as far as I can tell the auth.users table is protected and so I can't add policies to it. Is this correct? If so can you suggest a way where I can control the frontend creation of users with role-based access control?

Many thanks for a great product :)

**Comment by GaryAustin1** [comment, marked as answer]
You would create postgres "security definer" functions to do any manipulation of auth tables.  You would have to enforce your own security as needed in the function.  Then you would call them with an rpc call.

But if you want to create or delete users you probably want to consider having server side code or edge functions that can use the service_role and make auth api calls to create or delete users.  There is more to creating a user than just adding a row to the auth.users table.  
Create user API call https://supabase.com/docs/reference/javascript/auth-api-createuser

Here are just two threads of attempts at creating users by manipulating the auth schema and the complexity of that approach.
https://github.com/supabase/supabase/discussions/5043
https://github.com/supabase/supabase/discussions/5248
